jni/include/jni_util.h

```diff
@@ -200,13 +204,36 @@ namespace knn_jni {
         jfieldID GetFieldID(JNIEnv * env, jclass clazz, const char *name, const char *sig) final;
         jint CallNonvirtualIntMethodA(JNIEnv *env, jobject obj, jclass clazz, jmethodID methodID, jvalue *args) final;
         jlong CallNonvirtualLongMethodA(JNIEnv * env, jobject obj, jclass clazz, jmethodID methodID, jvalue* args) final;
+        void CallNonvirtualVoidMethodA(JNIEnv * env, jobject obj, jclass clazz, jmethodID methodID, jvalue* args) final;
         void * GetPrimitiveArrayCritical(JNIEnv * env, jarray array, jboolean *isCopy) final;
         void ReleasePrimitiveArrayCritical(JNIEnv * env, jarray array, void *carray, jint mode) final;
 
     private:
         std::unordered_map<std::string, jclass> cachedClasses;
         std::unordered_map<std::string, jmethodID> cachedMethods;
-    };
+    };  // class JNIUtil
+
+    /**
+     * It's common cleaner to release a primitive array within its destructor.
+     * Ex: JNIReleaseElements release_int_array_elements {[=](){
+     *   jniUtil->ReleaseIntArrayElements(env, idsJ, idsCpp, JNI_ABORT);
+     * }};
+     */
+    struct JNIReleaseElements {
+      explicit JNIReleaseElements(std::function<void()> _release_func)
+          : release_func(std::move(_release_func)) {
+      }
+
+      ~JNIReleaseElements() {
```

Comment:
The destructor should be marked noexcept to avoid any exceptions from propagating out, which is best practice for destructors.

Reply:
I don't think that is the best practice.
In c++, destructors are implicitly `noexcept(true)`. actually, even if we throw an exception there, the mark will still be `noexcept(true)`.

Reply:
I see from c++11 they made destructors implicit with noexcept(true), ..I was not aware of that !! we don't need this then .